testing/__init__.py

```diff
@@ -791,7 +791,7 @@ def __call__(self, *args, **kwargs):
     def _increment_counter(self, counter_key):
         # type: (str) -> int
         self._counters[counter_key] += 1
-        return self._counters[counter_key]
+        return cast(int, self._counters[counter_key])
```

Comment:
It would be better to type `self._counters = defaultdict(int)  # type: DefaultDict[str, int]` where you declare / initialize, but it would be arguably even better to use `collections.Counter` (with similar declaration site typing) which is exactly built for this.